controllers/opentelemetrycollector_controller.go

```diff
@@ -83,31 +83,31 @@ func (r *OpenTelemetryCollectorReconciler) onPlatformChange() error {
 	}
 	r.muTasks.Unlock()
 
-	if err := r.addRouteTask(plt, routesIdx); err != nil {
+	if err := r.addRouteTask(openshiftRoutesAvailable, routesIdx); err != nil {
 		return err
 	}
 
-	return r.removeRouteTask(plt, routesIdx)
+	return r.removeRouteTask(openshiftRoutesAvailable, routesIdx)
 }
 
-func (r *OpenTelemetryCollectorReconciler) addRouteTask(plt platform.Platform, routesIdx int) error {
+func (r *OpenTelemetryCollectorReconciler) addRouteTask(openshiftRoutesAvailable openshift_routes.OpenShiftRoutesAvailability, routesIdx int) error {
 	r.muTasks.Lock()
 	defer r.muTasks.Unlock()
 	// if exists and platform is openshift
-	if routesIdx == -1 && plt == platform.OpenShift {
+	if routesIdx == -1 && openshiftRoutesAvailable {
 		r.tasks = append([]Task{{reconcile.Routes, "routes", true}}, r.tasks...)
 	}
 	return nil
 }
 
-func (r *OpenTelemetryCollectorReconciler) removeRouteTask(plt platform.Platform, routesIdx int) error {
+func (r *OpenTelemetryCollectorReconciler) removeRouteTask(openshiftRoutesAvailable openshift_routes.OpenShiftRoutesAvailability, routesIdx int) error {
 	r.muTasks.Lock()
 	defer r.muTasks.Unlock()
 	if len(r.tasks) < routesIdx {
 		return fmt.Errorf("can not remove route task from reconciler")
 	}
 	// if exists and platform is not openshift
-	if routesIdx != -1 && plt != platform.OpenShift {
+	if routesIdx != -1 && openshiftRoutesAvailable {
```

Comment:
Do we not need to compare it against something like `openshift_routes.NotAvailable`? And we should change the comment on top.

Reply:
It is a bool so, it is not needed to compare to anything... but I agree it will be more meaningful.